### PR TITLE
implement phone number verification with SMS

### PR DESCRIPTION
**Pull request by marinovdeyan**
- SignUp view, UserViewModel and UserRespository updated to implement the phone verification of Firestore
- SMS sending currently not functional because we have no payment method in Firestore, but you can use the "testing phone numbers" feature
- no tests implemented yet, but feature seems functional with manual end-to-end test

@Sobell13 could you please have a look at what I've done so far and confirm to me that everything is in accordance with your implementation of UserViewModel and UserRepository(Firestore)?

I will not be merging this PR because it is currently untested (well, it works when I interact with it, but there are no unit and UI tests) and will definitely require polishing up, but I need some confirmation that my implementation fits with our current structure.

### CODE REVIEW

**Comment by marinovdeyan** [comment]
Tests have now been added to cover all the development in SignUp.kt (View), UserViewModel.kt and UserRepositoryFirestore (Model) for the sending and verifying of SMS codes.

What remains to be tested is actually sending real SMS on a real Android device (which I don't have - the sending of real SMSs is disabled on the emulator).

**Review by DanR2003 — APPROVED** [review]
The implementation of phone verification with Firestore looks well made:

- The work on SignUp view, UserViewModel, and UserRepository is well integrated and is thoughtfully implemented, even tough documentation is not very present.

- For the future, I think it would be a good idea to create separate viewModel files as the UserViewModel is becoming increasingly large and hard to read.

- It's nice to see that Firestore’s "testing phone numbers" feature was used as a workaround for SMS functionality, given the limitations. 

- SignUp.kt, UserViewModel.kt, and UserRepositoryFirestore look carefully tested.

 I think this PR is safe to merge into the main branch even tough SMS verification wasn't tested yet on an actual device, because the files have undergone carefull testing. 
 Tip for the future: Check that 2 users can't have the same phone number (it should be unique). At the moment we just check that the email is unique.

File: app/.attach_pid5305
Comment (DanR2003) [inline app/.attach_pid5305:1]: Should this empty file exist?

**Review by DanR2003 — APPROVED** [review]

**Review by ifaresse — CHANGES_REQUESTED** [review]
When an user try to register using a non test phone number you get this error : 
Capture d'écran 2024-11-11 174217
Otherwise the interface works well :
photo_2024-11-11_17-47-57 (2)
photo_2024-11-11_17-47-57
One good thing would be to have a limit of 6 characters in the verification code.

File: app/.attach_pid5305
Comment (marinovdeyan) [reply app/.attach_pid5305:1]: No, it's something that just appeared there without my knowledge. I will delete it.

Comment (marinovdeyan) [reply app/.attach_pid5305:1]: I have removed this in my latest push to this PR. Case closed.

**Review by ifaresse — APPROVED** [review]
LGTM

**Pull request merged**